posydon/binary_evol/DT/step_detached.py

```diff
@@ -605,328 +676,219 @@ def get_mist0(self, star, htrack):
-                                        " in the MIST model options.")
-                x0 = get_root0(MESA_label, posydon_attribute, htrack, rs=rs)
-                bnds = ([m_min_H, m_max_H], [0, None])
-                sol = minimize(
-                    lambda x: (
-                        (sc_he_core_mass_H.transform(
-                          get_track_val(MESA_label[0], htrack, *x))
-                         - sc_he_core_mass_H.transform(posydon_attribute[0]))
-                        )
-                    ** 2 + (sc_mass_H.transform(
-                        get_track_val(MESA_label[1], htrack, *x))
-                             - sc_mass_H.transform(posydon_attribute[1])
-                            ) ** 2
-                    + (sc_center_he4_H.transform(
-                        get_track_val(MESA_label[2], htrack, *x))
-                       - sc_center_he4_H.transform(posydon_attribute[2])) ** 2
-                    + (sc_log_R_H.transform(
-                        get_track_val(MESA_label[3], htrack, *x))
-                        - sc_log_R_H.transform(posydon_attribute[3])) ** 2,
-
-                    x0, method="TNC", bounds=bnds,
-                    )
+                list_for_matching = self.list_for_matching_postMS
+            elif star.state in LIST_ACCEPTABLE_STATES_FOR_HeStar:
+                list_for_matching = self.list_for_matching_HeStar
+
+            MESA_labels = list_for_matching[0]
+            posydon_attributes = posydon_attribute(MESA_labels, star)
+            rs = list_for_matching[1]
+            colscalers = list_for_matching[2]
+            bnds = []
+            for i in range(3, len(list_for_matching)):
+                bnds.append(list_for_matching[i])
+
+            if self.verbose or self.verbose == 1:
+                print("Matching attributes and their normalizations :",
+                      MESA_labels, rs)
+            for i in MESA_labels:
+                if i not in self.root_keys:
+                    raise Exception("Expected matching parameter not "
+                                    "added in the single star grid options.")
+
+            scales = []
+            for MESA_label, colscaler in zip(MESA_labels, colscalers):
+                scale_of_attribute = scale(MESA_label, htrack, colscaler)
+                scales.append(scale_of_attribute)
+
+            def square_difference(x):
+                result = 0.0
+                for (MESA_label, posydon_attr,
+                     colscaler, scale_of_that_MESA_label) in zip(
+                         MESA_labels, posydon_attributes, colscalers, scales):
+                    single_track_value = scale_of_that_MESA_label.transform(
+                        get_track_val(MESA_label, htrack, *x))
+                    posydon_value = scale_of_that_MESA_label.transform(
+                        posydon_attr)
+                    if MESA_label == "center_he4":
```

Comment:
Do we still keep the relative value for center_he4?

Reply:
I will comment it out